webapp/components/test-file-results.js

```diff
@@ -164,8 +161,17 @@ class TestFileResults extends WPTFlags(LoadingState(PathInfo(
           status: data && data.status,
           message: data && data.message,
         };
-        if (this.reftestAnalyzer) {
-          result.screenshots = data && data.screenshots;
+        if (this.reftestAnalyzer && data && data.screenshots) {
+          // Clone the data because we might modify it.
+          const screenshots = JSON.parse(JSON.stringify(data.screenshots));
```

Comment:
What is the type of `data.screenshots`? Is there a more straightforward way to clone it depending on type?

Reply:
It is a plain object (path1: sha1, path2: sha2, ...) deserialized from JSON.

Reply:
Could be copied with `Object.assign` then?

Reply:
TIL!

```suggestion
          const screenshots = Object.assign({}, data.screenshots);
```